File: sub-auth-service/src/main/java/org/github/auth/shiro/mapper/IShiroMapper.java
package org.github.auth.shiro.mapper;

import lombok.NonNull;
import org.apache.ibatis.annotations.Param;
import org.github.auth.shiro.model.UserDTO;
import org.github.mybatis.MyBatisMapper;
import org.github.shiro.AuthorInfo;

import javax.annotation.Nullable;
import java.util.List;

@MyBatisMapper
public interface IShiroMapper {
  @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull @Param("userId") Long userId);

  @Nullable UserDTO queryUserInfo(@NonNull @Param("username") String username);
}

File: sub-auth-service/src/main/java/org/github/auth/shiro/mapper/IShiroMapper.kt
package org.github.auth.shiro.mapper

import org.apache.ibatis.annotations.Param
import org.github.mybatis.MyBatisMapper
import org.github.shiro.AuthorInfo
import org.github.auth.shiro.model.UserDTO

@MyBatisMapper
interface IShiroMapper {
  fun queryAuthorInfo(@Param("userId") userId: Long): List<AuthorInfo>

  fun queryUserInfo(@Param("username") username: String): UserDTO?
}
